src/api/client.go

```diff
@@ -80,7 +81,35 @@ func NewClient(addr string) *Client {
 // Get makes a GET request to an endpoint and unmarshals the response to obj.
 // If the response is not 200 OK, returns an error
 func (c *Client) Get(endpoint string, obj interface{}) error {
-	resp, err := c.get(endpoint)
+	resp, err := c.get(endpoint, "")
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+
+		return NewClientError(resp.Status, resp.StatusCode, string(body))
+	}
+
+	if obj == nil {
+		return nil
+	}
+
+	d := json.NewDecoder(resp.Body)
+	d.DisallowUnknownFields()
+	return d.Decode(obj)
+}
+
+// GetJSON makes a GET request to an endpoint with Content-Type "application/json" and unmarshals the response to obj.
+// If the response is not 200 OK, returns an error
+func (c *Client) GetJSON(endpoint string, obj interface{}) error {
```

Comment:
There is no need for `GetJSON` since the API does not accept JSON bodies for GET requests